### DISCUSSION TITLE
Cloudflare closes websocket connection

### DISCUSSION THREAD

**Discussion by dgilev** [gravitational/teleport#11762] [Q&A] [answered]
Hello, 

I've got an issue after teleport was upgraded from 8.2.0 to 9.0.3. All of a sudden cloudflare (we use it to proxy requests to AWS ALB / teleport instances) started to terminate teleport SSH sessions/websocket connections after 100 seconds of inactivity which is expected if no keep-alives are sent from the server to client. Looks like it's happening because startPingLoop was completely removed from lib/web/terminal.go as a part of #8563 so no pings are being sent now. Is there any way to overcome this? 

Thanks!

**Comment by xacrimon** [comment, marked as answer]
Hi, this is fixed by https://github.com/gravitational/teleport/pull/11765 and will be put in a v9 bug release soon.
